dandiapi/api/services/publish/__init__.py

```diff
@@ -173,6 +173,7 @@ def _publish_dandiset(dandiset_id: int, user_id: int) -> None:
 
         # Set the version of the draft to PUBLISHED so that it cannot be published again without
         # being modified and revalidated
+        old_version.metadata['assetsSummary'] = new_version.metadata['assetsSummary']
         old_version.status = Version.Status.PUBLISHED
         old_version.save()
```

Comment:
The old comment needs to be moved to the next line, + I suggest adding a new comment along these lines describing why we're setting the assets summary
```suggestion
        # Synchronously update the assets summary of the draft version, since the
        # assets summary cron won't pick it up until it has been modified further.
        old_version.metadata['assetsSummary'] = new_version.metadata['assetsSummary']
        # Set the version of the draft to PUBLISHED so that it cannot be published again without
        # being modified and revalidated
        old_version.status = Version.Status.PUBLISHED
        old_version.save()
```